js/pulse.js

```diff
@@ -47,6 +47,7 @@
              */
             this.initialize = function(adManagerController, playerId) {
                 amc = adManagerController; // the AMC is how the code interacts with the player
+                amc.adManagerWillControlAds();
```

Comment:
Have you tested replaying the same video or watching videos back to back with this?  I don't think initialize will get run again. So the adManagerWillControlAds won't work the second time around.  If you would like an example, please refer to google_ima.js.  Inside that ad manager the adManagerWillControlAds is called when the ad manager tries to do the ad request.  That way, if you don't have an ad tag, you won't mess with the flow of the player.